### PR TITLE
Ignore compiled frontend code

### PR DESCRIPTION
**Pull request by SebastiaanKloos**
As the deploy script contains the compiling of the frontend code, I suggest adding these files to the gitignore file. What do you think @Cannonb4ll?

### CODE REVIEW

**Review by Glennmen — APPROVED** [review]

**Comment by Glennmen** [comment]
In order for this to work properly you also need to delete the existing files or else the old ones will remain in the repo and people would be pulling old versions.

**Comment by Cannonb4ll** [comment]
Problem here is that I don't compile on the server, I always compile locally and push that into GIT.

**Comment by Glennmen** [comment]
I understand that in the past this might have been fine for you but you might need to rethink that flow because this is now an open source project where multiple people can work on and contribute changes.

I can add a workflow to install npm packages and run mix to check if all assets stil compile correctly.

**Comment by Cannonb4ll** [comment]
I'm all open for you 2 to make this work as it should, I'll throw my OCD onto the side 😅

File: .gitignore [hunk 1]
```diff
@@ -12,5 +11,13 @@ _ide_helper.php
...
-/public/js/resources*.js
+/public/storage
 .phpunit.result.cache
+
+# Ignore compiled frontend code
+/public/js/frontend.js
+/public/js/app.js
+/public/css/frontend.css
+/public/css/app.css
+/public/js/resources*.js
```
Comment (Glennmen) [inline .gitignore:22]: Can't we ignore the directories?

```
/public/js/*
/public/css/*
```

Comment (SebastiaanKloos) [reply .gitignore]: I've changed that now. It works really well. Thanks!

**Comment by SebastiaanKloos** [comment]
@Glennmen @Cannonb4ll I've just fixed the workflow action to reflect the changes made to this branch

File: .github/workflows/php_test.yml [hunk 2]
```diff
@@ -34,6 +34,18 @@ jobs:
         
       - name: Generate key
         run: php artisan key:generate
+        
+      - name: Setup NPM
```
Comment (Glennmen) [inline .github/workflows/php_test.yml:38]: If I am correct this step isn't needed, node 14 is preinstalled in the Github container https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#installed-software

Comment (SebastiaanKloos) [reply .github/workflows/php_test.yml:38]: @Glennmen I see, good catch! I'm still learning every day🤓

Comment (SebastiaanKloos) [reply .github/workflows/php_test.yml]: @Glennmen Fixed it!

**Comment by Glennmen** [comment]
Not sure why there are conflicts 🤔

**Comment by SebastiaanKloos** [comment]
> Not sure why there are conflicts 🤔

**Comment by Glennmen** [comment]
@SebastiaanKloos How did you delete those files? Did you also delete them from git
```
git rm -r --cached .
git add -A
git commit -am 'Removing ignored files'
```

**Comment by Cannonb4ll** [comment]
Can you guys let me know once to merge and what the outcome is?

**Review by Glennmen — APPROVED** [review]

**Comment by Cannonb4ll** [comment]
Was I still needed for review or is this still WIP?

**Comment by Glennmen** [comment]
@Cannonb4ll It can be reviewed and merged.

**Pull request merged**